Fixes incorrectly cased namespace usages.

hostep opened the pull request:
### Description (*)
While playing around with phpstan on level 0 on the Magento codebase, these problems showed up (among many others, but those are for another day)

```
$ vendor/bin/phpstan analyse --level=0 lib | grep 'referenced with incorrect case'
 3929/3929 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

  10     Class Magento\Framework\DB\Select referenced with incorrect case: Magento\Framework\Db\Select.

$ vendor/bin/phpstan analyse --level=0 app/code/Magento | grep 'referenced with incorrect case'
 14097/14097 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

  116    Class Magento\GiftMessage\Model\Save referenced with incorrect case: Magento\Giftmessage\Model\Save.
  12     Class Magento\Sales\Model\Order\Creditmemo\Item referenced with incorrect case: Magento\Sales\Model\Order\CreditMemo\Item.
  28     Class Magento\Framework\DB\Select referenced with incorrect case: Magento\Framework\Db\Select.
  9      Class Magento\Sales\Model\Order\Creditmemo\Item referenced with incorrect case: Magento\Sales\Model\Order\CreditMemo\Item.
```

_Update 2025-01-02_: More cases were found then the ones listed above. This time not only by using phpstan, but by searching for those I already fixed which revealed more than phpstan was able to find (in comments for example).

This PR fixes these.
I don't think these caused real issues, I think composer's autoloader is gracefully enough to load those incorrectly cased namespaces, but let's make them correct anyways.

### Related Pull Requests

### Fixed Issues (if relevant)

### Manual testing scenarios (*)
1. Run `vendor/bin/phpstan analyse --level=0 {some-path} | grep 'referenced with incorrect case'` on all Magento code, it should not output anything (don't run this on the root the M2 codebase, otherwise it will take hours and might crash your computer if you don't have at least 32 GB of memory)
2. Test all affected code and make sure that everything keeps working as expected (I'm assuming the automated tests will take care of this)

### Questions or comments
I might fix (static) test failures if they don't turn out to be too complex to handle, but won't write new automated tests, because I think that's not relevant in scope of this PR.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)


### Resolved issues:
1. [x] resolves magento/magento2#38640: Fixes incorrectly cased namespace usages.

hostep committed:
Fixes incorrectly cased namespace usages.

hostep commented:
@magento run all tests

hostep committed:
Fix static tests.

hostep commented:
The failure of the semantic version check is in my opinion something to ignore, in practice it shouldn't cause an issue.

Pushed new commit to fix static tests.

@magento run all tests

engcom-Lima commented:
@magento run all tests

engcom-Charlie commented:
Hi @hostep,

Thank you for your contribution!

### :heavy_check_mark: QA Passed

In order to proceed further on this PR, I have followed the mentioned steps 
**Before:** :heavy_multiplication_x: 
<img width="1242" alt="image" src="https://github.com/magento/magento2/assets/51680850/77af9dc7-d498-4b06-93f6-9e9d0764ab69">


**After:** :heavy_check_mark:  
<img width="827" alt="image" src="https://github.com/magento/magento2/assets/51680850/c7852e87-2f0e-431d-ade0-a2e26d50ad23">


As the build is failing, currently moving it further in Extended Testing.

Thank you!

engcom-Charlie commented:
@magento run all tests

engcom-Charlie commented:
Raised internal approval [JIRA](https://jira.corp.adobe.com/browse/AC-11828) for the SVC failure. Moving this PR to Pending Approval now. We will proceed ahead on this once will get all the required approvals.

<img width="1471" alt="image" src="https://github.com/magento/magento2/assets/51680850/82099a31-7b84-449e-87f4-426fb3591e40">

engcom-Charlie commented:
@magento create issue

hostep commented:
@engcom-Charlie: any news?

engcom-Charlie commented:
Hi @hostep, as mentioned in [here](https://github.com/magento/magento2/pull/37574#issuecomment-2066372550), we are waiting for all required approvals for the SVC failure. Once will get it, we will inform here and proceed further on this.

Thank you for your contribution!!

hostep commented:
I know, but that was 8 months ago, can't you send a reminder in your approval ticket, maybe somebody forgot to check it?

hostep commented:
Note to self, found another violation in latest codebase in the `setup` directory:
```
$ vendor/bin/phpstan analyse -c ./dev/tests/static/testsuite/Magento/Test/Php/_files/phpstan/phpstan.neon --level=0 . | grep -B3 'incorrect case'
 20783/20783 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ ------------------------------------------------------------------------------------------------------
  Line   app/code/Magento/Sales/Test/Unit/Block/Adminhtml/Order/Create/Items/GridTest.php
 ------ ------------------------------------------------------------------------------------------------------
  118    Class Magento\GiftMessage\Model\Save referenced with incorrect case: Magento\Giftmessage\Model\Save.
--
 ------ ----------------------------------------------------------------------------------------------------------------------------
  Line   app/code/Magento/Tax/Block/Item/Price/Renderer.php
 ------ ----------------------------------------------------------------------------------------------------------------------------
  12     Class Magento\Sales\Model\Order\Creditmemo\Item referenced with incorrect case: Magento\Sales\Model\Order\CreditMemo\Item.
--
 ------ ------------------------------------------------------------------------------------------------
  Line   app/code/Magento/Tax/Model/Api/SearchCriteria/JoinProcessor/CalculationData.php
 ------ ------------------------------------------------------------------------------------------------
  28     Class Magento\Framework\DB\Select referenced with incorrect case: Magento\Framework\Db\Select.
--
 ------ ------------------------------------------------------------------------------------------------
  Line   lib/internal/Magento/Framework/Setup/Declaration/Schema/DataSavior/SelectGenerator.php
 ------ ------------------------------------------------------------------------------------------------
  10     Class Magento\Framework\DB\Select referenced with incorrect case: Magento\Framework\Db\Select.
--
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------
  Line   setup/src/Magento/Setup/Model/BasePackageInfo.php
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------
  8      Class Magento\Framework\Filesystem\Directory\ReadFactory referenced with incorrect case: Magento\Framework\FileSystem\Directory\ReadFactory.
```

Also note to self, make following adjustments, so running phpstan against the entire codebase runs quicker and without errors:
```diff
diff --git a/dev/tests/static/testsuite/Magento/Test/Php/_files/phpstan/phpstan.neon b/dev/tests/static/testsuite/Magento/Test/Php/_files/phpstan/phpstan.neon
index c6cf0a2a9c1..d356723e4c7 100644
--- a/dev/tests/static/testsuite/Magento/Test/Php/_files/phpstan/phpstan.neon
+++ b/dev/tests/static/testsuite/Magento/Test/Php/_files/phpstan/phpstan.neon
@@ -4,11 +4,12 @@ parameters:
     reportUnmatchedIgnoredErrors: false
     excludePaths:
         - %rootDir%/../../../lib/internal/Magento/Framework/ObjectManager/Test/Unit/*
+        - %rootDir%/../../../lib/internal/Magento/Framework/TestFramework/Unit/Listener/*
         - %rootDir%/../../../*/_files/*
-        - %rootDir%/../../../dev/tests/*/Fixtures/*
-        - %rootDir%/../../../dev/tests/*/tmp/*
-        - %rootDir%/../../../dev/tests/*/_generated/*
+        - %rootDir%/../../../dev/tests/*
+        - %rootDir%/../../../generated/*
         - %rootDir%/../../../pub/*
+        - %rootDir%/../../../vendor/*
     scanDirectories:
         - %rootDir%/../../../dev/tests/static/framework/tests/unit/testsuite/Magento
         - %rootDir%/../../../dev/tests/integration/framework/tests/unit/testsuite/Magento
```


Will try to fix that one in `setup` as well when I find a bit of time.

hostep commented:
Fixed more similar problems, have no more time to fix static tests, will follow up in a few days. No need to run tests, I'm aware that failures will happen.

hostep commented:
Found some more issues and fixed more static test failures. Then rebased interactively and squashed/splitted some commits together/out, so they are easier to understand. First commit solves all incorrect case-sensitive usages I was able to find, and the second commit deals with fixing of static test failures.

@magento run all tests

hostep commented:
@magento run all tests

hostep commented:
@magento run Static Tests, Integration Tests

hostep commented:
@magento run all tests

hostep commented:
The additional changes done to `app/code/Magento/Sales/Test/Unit/Block/Adminhtml/Order/Create/Items/GridTest.php` were to fix a complaint from php mess detector that the number of lines in a method shouldn't exceed 100, since I didn't touch anything originally in that method, I just removed some newslines, to get the number of lines under 100. I didn't want to refactor the whole thing, that is out of scope of this PR.

@magento run Database Compare

hostep committed:
Merge remote-tracking branch 'upstream/2.4-develop' into fixes-incorrectly-case-namespace-usages

hostep commented:
@magento run all tests

engcom-Charlie commented:
Hi @hostep,

I truly appreciate your patience.

As mentioned [above](https://github.com/magento/magento2/pull/37574#issuecomment-2566209855), we are following up on this, and once it gets approved, we'll proceed accordingly.

Once again, thank you for your contribution and patience!!

hostep commented:
@engcom-Charlie: can you explain why it takes over 15 months to get something small and silly like this to be approved? This should be a no-brainer ...

engcom-Charlie committed:
Merge branch '2.4-develop' into fixes-incorrectly-case-namespace-usages

hostep committed:
Also fixed incorrectly cased reference to the UploadResizeConfigValue viewmodel.

hostep commented:
Updated PR because I just found another wrongly cased reference in the file `app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/attribute/steps/bulk.phtml`

@magento run all tests

ihor-sviziev committed:
Merge branch '2.4-develop' into fixes-incorrectly-case-namespace-usages
